src/CPUSolver.cpp

```diff
@@ -900,44 +698,41 @@ void CPUSolver::scalarFluxTally(segment* curr_segment,
 
 
 /**
- * @brief Computes the exponential term in the transport equation for a
- *        Track segment.
- * @details This method computes \f$ 1 - exp(-l\Sigma^T_g/sin(\theta_p)) \f$
- *          for a segment with total group cross-section and for some polar
- *          angle. This method uses either a linear interpolation table
- *          (default) or the exponential intrinsic exp(...) function if
- *          requested by the user through a call to the
- *          Solver::useExponentialIntrinsic()  routine.
- * @param sigma_t the total group cross-section at this energy
- * @param length the length of the Track segment projected in the xy-plane
- * @param p the polar angle index
- * @return the evaluated exponential
+ * @brief Tallies the current contribution from this segment across the
+ *        the appropriate CMFD mesh cell surface.
+ * @param curr_segment a pointer to the Track segment of interest
+ * @param azim_index a pointer to the azimuthal angle index for this segment
+ * @param track_flux a pointer to the Track's angular flux
+ * @param fwd
  */
-FP_PRECISION CPUSolver::computeExponential(FP_PRECISION sigma_t,
-                                           FP_PRECISION length, int p) {
+void CPUSolver::tallySurfaceCurrent(segment* curr_segment,
+                                    int azim_index,
+                                    FP_PRECISION* track_flux,
+                                    bool fwd){
 
-  FP_PRECISION exponential;
-  FP_PRECISION tau = sigma_t * length;
-
-  /* Evaluate the exponential using the lookup table - linear interpolation */
-  if (_interpolate_exponential) {
-    int index;
-    index = round_to_int(tau * _inverse_exp_table_spacing);
-    index *= _two_times_num_polar;
-    exponential = (1. - (_exp_table[index+2 * p] * tau +
-                  _exp_table[index + 2 * p +1]));
-  }
+  FP_PRECISION surf_current;
 
-  /* Evalute the exponential using the intrinsic exp(...) function */
-  else {
-    FP_PRECISION sintheta = _polar_quad->getSinTheta(p);
-    exponential = 1.0 - exp(- tau / sintheta);
+  /* Tally surface currents if CMFD is in use */
+  if (_cmfd != NULL && _cmfd->isFluxUpdateOn()){
```

Comment:
Most segments will likely not intersect with a CMFD surface. I think it would be more computationally efficient to check if a segment intersects with a valid cmfd surface (i.e. `curr_segment->_cmfd_surface_fwd != -1` or `curr_segment->_cmfd_surface_bwd != -1` , depending on direction) before `surf_current` is computed, instead of checking in `Cmfd.cpp`.